tests/data/format/numpydoc/numpydoc_section_ordering.py.out

```diff
@@ -0,0 +1,63 @@
+"""Example module for numpydoc docstring style.
+References
+-----
+NumPy docstring style guide:
+https://numpydoc.readthedocs.io/en/latest/format.html#documenting-modules"""
+import math
+
+EULER_NUMBER = math.e
+"""Euler's number.
+
+Not related to Euler's constant (sometimes called the Euler-Mascheroni
+constant.
+Notes
+---
+It is the limit of ``(1 + 1/n)**n`` as n approaches infinity, so it
+is used in the equation for continuously-compouned interest.
+
+It is also the sum of the reciprocals of the whole numbers starting
+with zero, which is related to some calculus-related properties
+mathemeticians find elegant.
+
+References
+-----
+E (mathematical constant)
+https://en.wikipedia.org/wiki/E_(mathematical_constant)"""
+
+
+def sincos(theta):
+    """    Parameters
```

Comment:
This should probably be fixed.

Reply:
A variant on `textwrap.dedent` that ignored the indentation of the first line for determining common whitespace, which might work, then a similar variant of `textwrap.indent` at the end.

I would need to test how both handle
```python
    """\
    A summary line, indented the same as the others for all tools.

    Section Header
    --------------
    ...
    """
```

Another option might be to replace the `f"{quotes:s}{body:s}{quotes:s}"` reconstructing the new docstring with `f"{quotes:s}\\\n{body:s}{quotes:s}"` if the first character of the body is whitespace, but that still leaves the question of what to do with that "Returns" at the left margin.

Reply:
Can't we just check which line we are inserting on and if it's `0` we remove any indent and if higher we add an indent if it is missing?

Reply:
That's another option.  

The [numpydoc style guide](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) doesn't specify whether it prefers first line of docstring on same line as quotes or different line, but has [examples of both](https://numpydoc.readthedocs.io/en/latest/format.html#short-summary).  The [numpydoc validation tool](https://numpydoc.readthedocs.io/en/latest/validation.html#built-in-validation-checks) suggests the 
```python
    """\
    Summary.

    Extended summary...
    """
```
form is preferred to the
```python
    """Summary.

    Extended summary...
    """
```
form, which doesn't seem to agree with the style guide.  It also insists on extended summary and "See Also" sections, which I don't see always being necessary.

Supporting the first form is relatively straightforward; looping through the first lines of each section and doing `line.lstrip()` on the first section or `textwrap.indent(line)` for subsequent sections is also straightforward.